app/src/main/kotlin/edu/stanford/bdh/engagehf/bluetooth/BluetoothViewModel.kt

```diff
@@ -1,39 +1,73 @@
 package edu.stanford.bdh.engagehf.bluetooth
 
+import androidx.health.connect.client.records.HeartRateRecord
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import dagger.hilt.android.lifecycle.HiltViewModel
+import edu.stanford.bdh.engagehf.bluetooth.component.OperationStatus
 import edu.stanford.bdh.engagehf.bluetooth.data.mapper.BluetoothUiStateMapper
+import edu.stanford.bdh.engagehf.bluetooth.data.mapper.MeasurementToRecordMapper
+import edu.stanford.bdh.engagehf.bluetooth.data.models.Action
 import edu.stanford.bdh.engagehf.bluetooth.data.models.BluetoothUiState
+import edu.stanford.bdh.engagehf.bluetooth.data.models.MeasurementDialogUiState
+import edu.stanford.bdh.engagehf.bluetooth.data.models.VitalDisplayData
+import edu.stanford.bdh.engagehf.bluetooth.data.models.VitalDisplayUiState
+import edu.stanford.bdh.engagehf.bluetooth.data.repository.ObservationRepository
+import edu.stanford.healthconnectonfhir.RecordToObservationMapper
 import edu.stanford.spezi.core.bluetooth.api.BLEService
 import edu.stanford.spezi.core.bluetooth.data.model.BLEServiceEvent
 import edu.stanford.spezi.core.bluetooth.data.model.BLEServiceState
+import edu.stanford.spezi.core.bluetooth.data.model.Measurement
 import edu.stanford.spezi.core.logging.speziLogger
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asSharedFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
+import java.time.format.DateTimeFormatter
+import java.util.Locale
 import javax.inject.Inject
 
 @HiltViewModel
 class BluetoothViewModel @Inject internal constructor(
     private val bleService: BLEService,
     private val uiStateMapper: BluetoothUiStateMapper,
```

Comment:
here i'm not quite sure, because not all mappers have anything to do with the ui state. in the ui state we show the "measurement" and some mappers we only use to map the measurement into a record or observation so that we can save it, but don't use it for visualisation. or would you still include it in the ui state mapper, even if it's not directly related to the ui state, @eldcn ?